clorm/orm/core.py

```diff
@@ -2384,6 +2384,13 @@ def _generate_dynamic_predicate_functions(class_name: str, namespace: Dict):
         tmp.append(f"{f.name}_cltopy(raw_args[{idx}]), ")
     args_cltopy= "".join(tmp)
 
+    if pdefn.is_tuple:
```

Comment:
I'm not so familiar in defining an own hash-function but why we don't have to add the sign to the hash-function?

Reply:
You're right that the sign probably should be in the hash function. It doesn't matter in terms of correctness but would potentially reduce hash clashes.

This is another boundary case because negated facts are rarely used in well-written ASP programs. So I was doing a bit of premature optimisation :( by thinking that since it's rarely used it's cheaper to leave it out of the hash calculation. Which  is a bit crazy considering the tiny cost of adding it!